t/lib/t/MusicBrainz/Server/Controller/WS/js/Edit.pm

```diff
@@ -1209,6 +1209,85 @@ test 'undef relationship begin_date/end_date fields are ignored (MBS-8317)' => s
     is(scalar(@edits), 0);
 };
 
+test 'MBS-9063: Editing a relationship to use a type that does not support dates clears them' => sub {
+    my $test = shift;
+    my ($c, $mech) = ($test->c, $test->mech);
+
+    prepare_test_database($c);
+
+    $mech->get_ok('/login');
+    $mech->submit_form( with_fields => { username => 'new_editor', password => 'password' } );
+
+    my $edit_data = [ {
+        edit_type   => $EDIT_RELATIONSHIP_EDIT,
+        id          => 66666123,
+        linkTypeID  => 834,
+        attributes  => [],
+        entities => [
+            { gid => '1e6092a0-73d3-465a-b06a-99c81f7bec37' },
+            { gid => '145c079d-374e-4436-9448-da92dedef3cf' },
+        ],
+        begin_date   => { year => 1999, month => 1, day => 1 },
+        end_date     => { year => 2009, month => 9, day => 9 },
+        ended       => 1,
```

Comment:
I was thinking it'd make sense for this to be an error--if dates are submitted where they aren't supported, then I'm not sure we should make any assumptions (e.g., the link type could be wrong instead). What do you think?

At the same time, I'd suggest we should be able to omit the `begin_date`, `end_date`, and `ended` properties entirely if dates aren't supported, so this should probably be two tests if the former is an error.

Reply:
Hmm. When are you thinking we would submit the wrong link type through this internal use API? I *guess* we could error, but it feels submitting the wrong link type would be a strange fuckup to make in our own code?

Reply:
I was more thinking about the user selecting the wrong one accidentally, and there being a bug in the code where we aren't clearing them from the JS side. But that would be rare and probably annoying for the user.

Reply:
I honestly feel in that case it would be still clearer if the change was made and the dates dropped, rather than giving an error about dates being passed that the user had nothing to do with since they either expected dates to be passed and why is it an error, or expected them to be dropped and how is it their fault.